pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -192,19 +175,20 @@ type etcd struct {
 	role                    string
 	class                   Class
 	annotations             map[string]string
+	failureToleranceType    *gardencorev1beta1.FailureToleranceType
 	replicas                *int32
 	storageCapacity         string
 	defragmentationSchedule *string
-	nodeSpread              bool
-	zoneSpread              bool
+	caRotationPhase         gardencorev1beta1.ShootCredentialsRotationPhase
 	k8sVersion              string
+	etcd                    *druidv1alpha1.Etcd
+	backupConfig            *BackupConfig
+	hvpaConfig              *HVPAConfig
+	ownerCheckConfig        *OwnerCheckConfig
+}
 
-	etcd *druidv1alpha1.Etcd
-
-	backupConfig     *BackupConfig
-	hvpaConfig       *HVPAConfig
-	ownerCheckConfig *OwnerCheckConfig
-	caRotationPhase  gardencorev1beta1.ShootCredentialsRotationPhase
+func (e *etcd) hasHAControlPlane() bool {
+	return helper.IsFailureToleranceTypeNode(e.failureToleranceType) || helper.IsFailureToleranceTypeZone(e.failureToleranceType)
```

Comment:
Should we determine this once in the constructor and assign the result to a `var` instead of calling the function again and again?

Reply:
The cost of invoking this method is negligible and it prevents polluting the etcd struct which already has `failureToleranceType`. Compiler also optimizes these method calls. I do not see an issue here.